### ISSUE TITLE
Using the ArtifactoryDocker task - [Error] Could not find docker image in Artifactory, expecting image ID:

### ISSUE THREAD

**Issue by carterkraft** [jfrog/jfrog-cli#803]
**Describe the bug**

Using the ArtifactoryDocker@1 task with push command in a Azure DevOps Pipeline with collectBuildInfo set to true

20200811.6: digest: sha256:8b079337c0016e5d6d630421c425789014fcb6b948ee4b00501f71f93a18db28 size: 3248
[Info] Getting repository 'it-builds' details ...
[Info] Done getting repository details.
**[Error] Could not find docker image in Artifactory, expecting image ID:**
##[error]Error: Command failed: /azp/agent/_work/_tool/jfrog/1.37.1/x64/jfrog rt dp "<REMOVED_REPO_URL>/azdo/pipeline/it-docker-build-agent:20200811.6" "it-builds" --url="<REMOVED_REPO_URL>" --user=*** --password=***
Finishing: ArtifactoryDocker

**To Reproduce**
Using the ArtifactoryDocker@1 task with push command in a Azure DevOps Pipeline with collectBuildInfo set to true

**Expected behavior**
As of last Friday August 7th this successfully worked.

Here is the log from that build, which was the same commit as of today August 11th it now fails: 

[Info] Getting repository 'it-builds' details ...
[Info] Done getting repository details.
[Info] Setting properties...
[Info] [Thread 0] Setting properties on: <URL REMOVED>/latest/manifest.json
[Info] [Thread 1] Setting properties on: <URL REMOVED>/latest/sha256__80337af41fb252c6c22bcd22b3cece43b45c374ceda0839225d337df69fa64c1
[Info] [Thread 2] Setting properties on: <URL REMOVED>/latest/sha256__8559a31e96f442f2c7b6da49d6c84705f98a39d8be10b3f5f14821d0ee8417df
[Info] [Thread 1] Setting properties on: <URL REMOVED>/latest/sha256__350223dc6418fa8a7ff26f3b1bf1ed555c120d5aec7548f4bc51c26dc7e3d568
[Info] [Thread 2] Setting properties on: <URL REMOVED>/latest/sha256__08c5d69e6f61aa0d117ba2b7e26c7642300aa0b9a41ef9083cd5086463ccc5a3
[Info] [Thread 0] Setting properties on: <URL REMOVED>/latest/sha256__3d3f2533680d72a56366d3ce7edccc4cf8f3991229e13a4ddeb36fe87d886f2c
[Info] [Thread 1] Setting properties on: <URL REMOVED>/latest/sha256__541a6488fa852b35c20b292f695ba89d1705fe4b8d1a1b56f9e2754dc47e839c
[Info] [Thread 2] Setting properties on: <URL REMOVED>/latest/sha256__1a84c2e7a2555733cd835e3d778b0932809e07d78403f8f12a2240c3f14c852b
[Info] [Thread 0] Setting properties on: <URL REMOVED>/latest/sha256__2825ab9b7d6d8e5e5796bf5befa65d5cf53dbb164c04abe83864db5c07665c92
[Info] [Thread 1] Setting properties on: <URL REMOVED>/latest/sha256__a6116923d4e17f9980f7e2c115c7f509ed38f682bec4e142ef9de4609b2d3fe2
[Info] [Thread 0] Setting properties on: <URL REMOVED>/latest/sha256__6e836f2b5681784e08a86610f6db22dd57a86b6df6edb099dfc73d2f4875277c
[Info] [Thread 2] Setting properties on: <URL REMOVED>/latest/sha256__0286824399184c76f223d4b9f086b32b78cbc68af754d06cefa5c266b53a0958
[Info] [Thread 1] Setting properties on: <URL REMOVED>/latest/sha256__f0f12747544ab172b1f27b4da559e8183b447c062769b857ac65cae1be7cc236
[Info] [Thread 2] Setting properties on: <URL REMOVED>/latest/sha256__1b7c6330b34bab7488a9fa335111af99632e5f2b60c03fd082aee1e36c288f58
[Info] [Thread 0] Setting properties on: <URL REMOVED>/latest/sha256__dbaf814fb1f02875edd9dbd0f932a831d74b601988a6d530ec0cc8b2fc084628
[Info] [Thread 1] Setting properties on: <URL REMOVED>/latest/sha256__9296293ae7c28c7dc66973d01f5cef9b108e53bf3ba6fe837856aae86d81f1cf
[Info] Done setting properties.
Finishing: ArtifactoryDocker

**Screenshots**

![image](https://user-images.githubusercontent.com/37340374/89953202-3a9b9780-dbf4-11ea-8754-a0f8bf9775c2.png)

**Versions**
- JFrog CLI version: 1.37.1
- JFrog CLI operating system: Linux - Debian
- Artifactory Version: 6.10.1

**Additional context**
All the pipelines that used the the ArtifactoryDocker@1 task in a Azure DevOps Pipeline with collectBuildInfo set to true

**Comment by eyalbe4** [comment]
@carterkraft,
Could it be that the docker image is pushed to a different repository than the repository configured in the task? This can happen if the docker URL points to a different repository. The target repository for this URL is configured outside of Azure DevOps, in the reverse proxy in most cases. 
You can actually go to the Artifactory UI (or use JFrog CLI's search command) to see where the image was pushed to.
Please let us know what you see there.

**Comment by carterkraft** [comment]
@eyalbe4  - Thanks for taking the time to review and reply to my issue.  

We only configure the Artifactory Service connection one time in a task and the push is working successfully its when it tries to collect the build info from the same build it just pushed that it fails. 

I can see where the image was pushed and it was where I expected and was successful. Disabling the option to collect build info allows it to successfully complete the step.

**Comment by eyalbe4** [comment]
@carterkraft,

Yes - I understand that the build-info collection part fails, not the push.
My question is this - are you absolutely sure that the image you see in the repo configured in the **Artifactory Docker** task was indeed pushed by the task that failed to collect the build-info?
Are you sure this is not an older image?

Another question - the following log message you shared:
**[Error] Could not find docker image in Artifactory, expecting image ID:**
seems to be missing the actual image ID at the end. Can you please check that the end wasn't accidentally trimmed? The error message indicates that the build-info generation failed, because the manifest.json of the pushed image was not found in the configured repo. Here's the code that throws this error -  
https://github.com/jfrog/jfrog-cli/blob/master/artifactory/utils/docker/buildinfo.go

Please let me know what you see.

**Comment by eyalbe4** [comment]
Hi @carterkraft,
Did you get a chance to look into this?
Thanks!

**Comment by 1633605** [comment]
@carterkraft I have same issue while pushing docker image with publish build info as true. Have you figured out any solution or work around fix for this?
